Tests/NIOHTTP1Tests/HTTPDecoderTest.swift

```diff
@@ -917,4 +917,79 @@ class HTTPDecoderTest: XCTestCase {
             XCTAssertEqual(.invalidEOFState, error as? HTTPParserError)
         }
     }
+
+    func testDecodingLongHeaderFieldNames() {
+        XCTAssertNoThrow(try self.channel.pipeline.addHandler(ByteToMessageHandler(HTTPRequestDecoder())).wait())
+
+        var buffer = ByteBuffer(string: "GET / HTTP/1.1\r\nHost: example.com\r\n")
+
+        XCTAssertNoThrow(try self.channel.writeInbound(buffer))
+        XCTAssertNoThrow(XCTAssertNil(try self.channel.readInbound(as: HTTPServerRequestPart.self)))
+
+        buffer.clear()
+        buffer.writeRepeatingByte(UInt8(ascii: "x"), count: 1024)
+
+        for _ in 0..<80 {
```

Comment:
nit: 80 and 1024 are a bit magic here, can you comment with a reference to `maximumHeaderFieldSize` in these tests?